src/Documentation/Documentation.php

```diff
@@ -23,17 +23,13 @@
 final class Documentation
 {
     private $resourceNameCollection;
-    private $title;
-    private $description;
-    private $version;
+    private $info;
     private $mimeTypes = [];
 
-    public function __construct(ResourceNameCollection $resourceNameCollection, string $title = '', string $description = '', string $version = '', array $formats = [])
+    public function __construct(ResourceNameCollection $resourceNameCollection, array $info = [], array $formats = [])
     {
         $this->resourceNameCollection = $resourceNameCollection;
```

Comment:
As mentioned before, this would be a BC-break, what I suggest is doing this instead:

```suggestion
    public function __construct(ResourceNameCollection $resourceNameCollection, $titleOrInfo = '', string $description = '', string $version = '', array $formats = [])
    {
        $this->resourceNameCollection = $resourceNameCollection;
        if (\is_array($titleOrInfo)) {
          $this->info = $titleOrInfo;
        } else {
            // trigger a deprecation here : 'passing non-array as the third argument is deprecated...'
            $this->info = [
              'title' => $titleOrInfo,
              'description' => $description,
              'version' => $version,
            ];
        }
```